tests/devices/unit_tests/test_synchrotron.py

```diff
@@ -0,0 +1,223 @@
+]
+
+CONFIG_ADDRESSES = [
+    f"sim://{PV.STATUS_PREFIX + PV.BEAM_ENERGY}",
+    f"sim://{PV.STATUS_PREFIX + PV.MODE}",
+]
+
+READ_SIGNALS = [
+    "synchrotron-ring_current",
+    "synchrotron-machine_user_countdown",
+    "synchrotron-topup_start_countdown",
+    "synchrotron-top_up_end_countdown",
+]
+
+CONFIG_SIGNALS = [
+    "synchrotron-beam_energy",
+    "synchrotron-synchrotron_mode",
+]
+
+EXPECTED_READ_RESULT = f"""{{
+  "{READ_SIGNALS[0]}": {{
+    "{READING_FIELDS[0]}": {READINGS[0]},
+    "{READING_FIELDS[1]}": 0
+  }},
+  "{READ_SIGNALS[1]}": {{
+    "{READING_FIELDS[0]}": {READINGS[1]},
+    "{READING_FIELDS[1]}": 0
+  }},
+  "{READ_SIGNALS[2]}": {{
+    "{READING_FIELDS[0]}": {READINGS[2]},
+    "{READING_FIELDS[1]}": 0
+    }},
+  "{READ_SIGNALS[3]}": {{
+    "{READING_FIELDS[0]}": {READINGS[3]},
+    "{READING_FIELDS[1]}": 0
+  }}
+}}"""
+
+EXPECTED_READ_CONFIG_RESULT = f"""{{
+  "{CONFIG_SIGNALS[0]}":{{
+    "{READING_FIELDS[0]}": {CONFIGS[0]},
+    "{READING_FIELDS[1]}": 0
+  }},
+  "{CONFIG_SIGNALS[1]}":{{
+    "{READING_FIELDS[0]}": "{CONFIGS[1]}",
+    "{READING_FIELDS[1]}": 0
+  }}
+}}"""
+
+EXPECTED_DESCRIBE_RESULT = f"""{{
+  "{READ_SIGNALS[0]}":{{
+    "{DESCRIPTION_FIELDS[0]}": "{READING_ADDRESSES[0]}",
+    "{DESCRIPTION_FIELDS[1]}": "{NUMBER}",
+    "{DESCRIPTION_FIELDS[2]}": {EMPTY_LIST}
+  }},
+  "{READ_SIGNALS[1]}":{{
+    "{DESCRIPTION_FIELDS[0]}": "{READING_ADDRESSES[1]}",
+    "{DESCRIPTION_FIELDS[1]}": "{NUMBER}",
+    "{DESCRIPTION_FIELDS[2]}": {EMPTY_LIST}
+  }},
+   "{READ_SIGNALS[2]}":{{
+    "{DESCRIPTION_FIELDS[0]}": "{READING_ADDRESSES[2]}",
+    "{DESCRIPTION_FIELDS[1]}": "{NUMBER}",
+    "{DESCRIPTION_FIELDS[2]}": {EMPTY_LIST}
+  }},
+  "{READ_SIGNALS[3]}":{{
+    "{DESCRIPTION_FIELDS[0]}": "{READING_ADDRESSES[3]}",
+    "{DESCRIPTION_FIELDS[1]}": "{NUMBER}",
+    "{DESCRIPTION_FIELDS[2]}": {EMPTY_LIST}
+  }}
+}}"""
+
+EXPECTED_DESCRIBE_CONFIG_RESULT = f"""{{
+  "{CONFIG_SIGNALS[0]}":{{
+    "{DESCRIPTION_FIELDS[0]}": "{CONFIG_ADDRESSES[0]}",
+    "{DESCRIPTION_FIELDS[1]}": "{NUMBER}",
+    "{DESCRIPTION_FIELDS[2]}": {EMPTY_LIST}
+  }},
+  "{CONFIG_SIGNALS[1]}":{{
+    "{DESCRIPTION_FIELDS[0]}": "{CONFIG_ADDRESSES[1]}",
+    "{DESCRIPTION_FIELDS[1]}": "{STRING}",
+    "{DESCRIPTION_FIELDS[2]}": {EMPTY_LIST}
+  }}
+}}"""
+
+
+@pytest.fixture
+async def sim_synchrotron() -> OASynchrotron:
+    async with DeviceCollector(sim=True):
+        sim_synchrotron = OASynchrotron()
+    set_sim_value(sim_synchrotron.ring_current, RING_CURRENT)
+    set_sim_value(sim_synchrotron.machine_user_countdown, USER_COUNTDOWN)
+    set_sim_value(sim_synchrotron.topup_start_countdown, START_COUNTDOWN)
+    set_sim_value(sim_synchrotron.top_up_end_countdown, END_COUNTDOWN)
+    set_sim_value(sim_synchrotron.beam_energy, BEAM_ENERGY)
+    set_sim_value(sim_synchrotron.synchrotron_mode, MODE)
+    return sim_synchrotron
+
+
+async def test_oasynchrotron_read(sim_synchrotron: OASynchrotron):
```

Comment:
Must: All these tests need marking with `@pytest.mark.asyncio` otherwise pytest won't pick them up. I suspect this is why codecov is upset

Reply:
Callum put ayncio=auto in the pyproject.toml file (which I realise i haven't included in the push yet) so that the annotation wasn't necessary, is thatttt sufficient (assuming I include it)?

Reply:
I don't know, I guess so. In which case we should remove all the asyncio marks from elsewhere. Sounds like a new PR though. Has Callum put that change in a separate one?

Reply:
Keith is away afternoons this week and all next, I'm happy to isolate and push that change (I'm blocked waiting on p38 hardware)

Reply:
Joseph say the above is merged so should work now